src/coreclr/jit/lsrabuild.cpp

```diff
@@ -3207,8 +3207,24 @@ void LinearScan::BuildStoreLocDef(GenTreeLclVarCommon* storeLoc,
             srcInterval->assignRelatedInterval(varDefInterval);
         }
     }
-    RefPosition* def =
-        newRefPosition(varDefInterval, currentLoc + 1, RefTypeDef, storeLoc, allRegs(varDsc->TypeGet()), index);
+
+    regMaskTP defCandidates = RBM_NONE;
+    var_types type          = varDsc->TypeGet();
+
+#ifdef TARGET_X86
+    if (varTypeIsByte(type))
```

Comment:
would it be correct to add this logic into `allRegs(type)` instead?

Reply:
That sounds to be appropriate fix, but currently there are too many usage of `allRegs(type)` and I am not sure how that suggestion would impact other flows. My recommendation is to get this fix and generalize it in future if we find more cases like this.

Reply:
Could you please add an assert to `allRegs(type)` that `type != TYP_BYTE` to see if any other calls are impacted, if it fails then it should show other existing issues.